src/api/list.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include "mamba/api/list.hpp"
+#include "mamba/api/install.hpp"
+
+#include "mamba/core/channel.hpp"
+#include "mamba/api/configuration.hpp"
+#include "mamba/core/prefix_data.hpp"
+
+
+namespace mamba
+{
+    void list(const std::string& regex)
+    {
+        auto& config = Configuration::instance();
+
+        config.at("show_banner").get_wrapped<bool>().set_value(false);
+        config.load(MAMBA_ALLOW_ROOT_PREFIX | MAMBA_ALLOW_FALLBACK_PREFIX
+                    | MAMBA_ALLOW_EXISTING_PREFIX | MAMBA_ALLOW_MISSING_PREFIX
+                    | !MAMBA_ALLOW_NOT_ENV_PREFIX | MAMBA_EXPECT_EXISTING_PREFIX);
```

Comment:
again !MAMBA ... which is difficult to interpret

Reply:
will work on that